pyanaconda/async_utils.py

```diff
@@ -0,0 +1,86 @@
+# Utilities for asynchronous calls.
+#
+# Copyright (C) 2017  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+import gi
+
+from queue import Queue
+from pyanaconda.threading import threadMgr
+
+gi.require_version("GLib", "2.0")
+
+from gi.repository import GLib
```

Comment:
I am thinking if we could wrap all GLib functions we use in anaconda, so we don't have to do the crazy import every time. Just an idea for another issue.

Reply:
I have this as a future plan. We really need a wrapper for the GLib.